app/react/Auth/components/NeedAuthorization.tsx

```diff
@@ -1,24 +1,55 @@
-import React from 'react';
+import React, { useMemo } from 'react';
 import { connect } from 'react-redux';
+import { ObjectIdSchema } from 'shared/types/commonTypes';
+import { PermissionSchema } from 'shared/types/permissionType';
 
 type PropTypes = {
-  authorized: boolean;
   children: React.ReactNode;
+  roles?: string[];
+  orWriteAccessTo?: any[];
+  user: any;
 };
 
-const NeedAuthorization: React.FC<PropTypes> = ({ authorized, children }: PropTypes) =>
-  authorized ? <>{children}</> : null;
+const checkWritePermissions = (entities: any[] = [], user?: any) => {
```

Comment:
@daneryl At the UI level you cannot see things that you shouldn't see because they're not given to you from the server, so there's so a 'see permission' doesn't make sense in the front. This checks for write permissions, and the use case is when you select multiple entities and you need to discern whether you can bulk edit them or not.

Reply:
i still think this is not good UX but that is for another discussion. i understand the usecase